### PR TITLE
Finance connector + subgraph

### PR DESCRIPTION
**Pull request by 0xGabi**
Add a new connector for the Finance app with the new `connectApp()` API.

The first commit include the subgraph and a bootstrap for the connector repo. 
The second commit includes the logic of the Finance app connector.
The last commit includes a few tests.

### CODE REVIEW

**Comment by 0xGabi** [comment]
Not sure why GitHub closes this automatically. Doing a rebase from @bpierre branch so I'll open a new PR.

File: packages/connect-finance/src/entities/Finance.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  async disconnect() {
+    await this.#connector.disconnect()
+  }
+
+  async transactions({ first = 1000, skip = 0 } = {}): Promise<Transaction[]> {
+    return this.#connector.transactionsForApp(this.#appAddress, first, skip)
+  }
+
+  onTransactions(callback: Function): SubscriptionHandler {
```
Comment (bpierre) [inline packages/connect-finance/src/entities/Finance.ts:23]: I updated the voting connector recently, to align the parameters of `votes` with `onVotes` (https://github.com/aragon/connect/pull/135/commits/a2dbb76c1c5b132d1c46cb793bd1acef19007408). I think we could do the same here!

File: packages/connect-finance/src/entities/Finance.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    { first = 1000, skip = 0 } = {}
+  ): Promise<TokenBalance> {
+    return this.#connector.balanceForToken(
+      this.#appAddress,
+      tokenAddress,
+      first,
+      skip
+    )
+  }
+}
```
Comment (bpierre) [inline packages/connect-finance/src/entities/Finance.ts:38]: Should we add `onBalance()`?

File: packages/connect-finance/src/entities/TokenBalance.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+export interface TokenBalanceData {
+  id: string
+  token: Address
+  balance: string
+}
+
+export default class TokenBalance implements TokenBalanceData {
+  readonly id!: string
+  readonly token!: string
+  readonly balance!: string
```
Comment (bpierre) [inline packages/connect-finance/src/entities/TokenBalance.ts:12]: I don’t think the `!` are needed here?

File: packages/connect-finance/src/entities/Transaction.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+export default class Transaction implements TransactionData {
+  readonly id!: string
+  readonly token!: Address
+  readonly entity!: Address
+  readonly isIncoming!: boolean
+  readonly amount!: string
+  readonly date!: string
+  readonly reference!: string
```
Comment (bpierre) [inline packages/connect-finance/src/entities/Transaction.ts:20]: I think we could remove the `!` here as well.

File: packages/connect-finance/subgraph/README.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# Example Subgraph
+
+An example to help you get started with The Graph. For more information see the docs on https://thegraph.com/docs/.
```
Comment (bpierre) [inline packages/connect-finance/subgraph/README.md:3]:
```suggestion
# Finance Subgraph

Subgraph for the Finance app.
```

**Pull request merged**